Kontent.Ai.Delivery/ContentItems/DateTimeContentConverter.cs

```diff
@@ -0,0 +1,43 @@
+﻿using Kontent.Ai.Delivery.Abstractions;
+using Kontent.Ai.Delivery.ContentItems.DateTimes;
+using System;
+using System.Reflection;
+using System.Threading.Tasks;
+
+namespace Kontent.Ai.Delivery.ContentItems
+{
+    internal class DateTimeContentConverter : IPropertyValueConverter<DateTime?>
+    {
+        public DateTimeContentConverter()
+        {
+        }
+
+        public Task<object> GetPropertyValueAsync<TElement>(
+            PropertyInfo property,
+            TElement contentElement,
+            ResolvingContext context
+            ) where TElement : IContentElementValue<DateTime?>
+        {
+            if (!typeof(IDateTimeContent).IsAssignableFrom(property.PropertyType))
+            {
+                throw new InvalidOperationException($"Type of property {property.Name} must implement {nameof(IDateTimeContent)} in order to receive datetime content.");
+            }
+
+            if (contentElement is not IDateTimeElementValue element)
+            {
+                return Task.FromResult((object)null);
```

Comment:
Here is an interesting design decision to make. What should happen if you pass in the wrong element type?

PS: I see that other converters have chosen the same solution. 

Take this note as something to think about. No action is required.